File: Python/392.py
# Is Subsequence
# Easy

def isSubsequence(s: str, t: str) -> bool:

    subseq_char = 0
    subseq_size = len(s)
    if subseq_size == 0: return True

    for letter in t:
        if letter == s[subseq_char]:
            subseq_char += 1
            if subseq_char == subseq_size:
                return True
    return False

File: C++/392.cpp
// Is Subsequence
// Easy

#include <string>

using namespace std;

bool isSubsequence(string s, string t) {
        
    int subseq_char = 0;
    int subseq_size = s.length();
    if (subseq_size == 0) return true;
    for(auto &l : t) {
        if ( l == s[subseq_char] ){
            subseq_char ++;
            if ( subseq_char == subseq_size ) {
                return true;
            }
        }
    }
    return false;
}